File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet();
        
        for(int num:nums){
            set.add(num);
        }
        
        int max_sequence=0;
        
        for(int i=0;i<nums.length;i++){
            int current=nums[i];
            int current_seq_length=1;
        
            if(!set.contains(current-1)){
                while(set.contains(current+1)){
                    current+=1;
                    current_seq_length+=1;
                }
                max_sequence=Math.max(max_sequence,current_seq_length);
            }
        }
        return max_sequence;
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
         # Convert list to set for O(1) lookups
        s = set(nums)
        longest = 0

        for num in s:
            # Only start counting if num is the start of a sequence
            if num - 1 not in s:
                next_num = num + 1
                length = 1

                # Expand the sequence
                while next_num in s:
                    length += 1
                    next_num += 1

                # Update the longest length
                longest = max(longest, length)

        return longest
        